src/api_module.js

```diff
@@ -8,194 +8,305 @@ const fetchRetry = useFetchRetry(fetch, {
-              "x-secret": auth.secret,
-              "x-device-id": auth.device_id,
-            },
-          }
-        );
-      } catch (error) {
-        const reason = {};
-        reason[error.constructor.name] = [error.message];
-        throw reason;
-      }
-      const result = await response.json();
-      const loaded = new Date();
-      for (let obj in result) {
-        result[obj].loaded = loaded;
-      }
-      if (response.ok && result) {
-        if (response.headers.get("x-total-pages") === String(page)) {
-          return result;
-        } else {
-          yield result;
-        }
-      } else {
-        throw result;
-      }
+  index(auth) {
+    return indexGenerator(this.path, auth, new Scope());
+  }
 
-      page++;
-    }
+  async create(auth, object) {
+    return await httpPost(this.path, auth, object);
+  }
+
+  async read(auth, id, retryOptions = {}) {
+    return await httpGet(`${this.path}/${id}`, auth, retryOptions);
+  }
+
+  async destroy(auth, id) {
+    return await httpDelete(`${this.path}/${id}`, auth);
   }
 }
 
-export class RescanModule extends ApiModule {
+class CRUDModule extends CRDModule {
   constructor(path) {
-    super(path, []);
-  }
-
-  start = async function (auth, object) {
-    const request = new Request(`${this.path}`, {
-      method: "POST",
-      headers: {
-        "content-type": "application/json",
-        "x-secret": auth.secret,
-        "x-device-id": auth.device_id,
-      },
-      body: JSON.stringify(object),
-    });
-    return await this._resolveRequest(request);
-  };
-
-  show = async function (auth, retryOptions = {}) {
-    const request = new Request(`${this.path}`, {
-      ...retryOptions,
-      method: "GET",
-      headers: {
-        "x-secret": auth.secret,
-        "x-device-id": auth.device_id,
-      },
-    });
-    return await this._resolveRequest(request);
-  };
+    super(path);
+  }
+
+  async update(auth, id, object) {
+    return await httpPatch(`${this.path}/${id}`, auth, object);
+  }
+}
+
+export class AlbumModule extends CRUDModule {
+  constructor(baseURL) {
+    super(`${baseURL}/albums`);
+  }
+
+  index(auth, scope = new AlbumsScope()) {
+    return indexGenerator(this.path, auth, scope);
+  }
+
+  async destroyEmpty(auth) {
+    return await httpPost(`${this.path}/destroy_empty`, auth, {});
+  }
+}
+
+export class ArtistModule extends CRUDModule {
+  constructor(baseURL) {
+    super(`${baseURL}/artists`);
+  }
+
+  index(auth, scope = new ArtistsScope()) {
+    return indexGenerator(this.path, auth, scope);
+  }
+
+  async destroyEmpty(auth) {
+    return await httpPost(`${this.path}/destroy_empty`, auth, {});
+  }
+
+  async merge(auth, id, sourceId) {
+    return await httpPost(
+      `{this.path}/${id}/merge?source_id=${sourceId}`,
+      auth,
+      {}
+    );
+  }
+}
+
+export class AuthTokenModule extends CRDModule {
+  constructor(baseURL) {
+    super(`${baseURL}/auth_tokens`);
+  }
+
+  async create(object) {
```

Comment:
The readme should be updated to reflect this

Reply:
Done.